Bottom: 9fbd5a9eb19f822e1fd393f1d441c7b75aa9b605
Top:    5b15f5f7f4b37aba2e62a8a857486c6e41f1375f
Author: Parth Shah <parth@linux.ibm.com>
Date:   2019-10-07 14:00:46 +0530

sched/core: Add manual background task classification using sched_setattr syscall

Small background tasks typically performs some housekeeping work and are
less important in the overall scheme of load balancing and scheduling.

So provide a way to mark the task which are small background noises with
the use of additional flag to the existing task attribute. Also provide an
interface from the userspace which uses sched_setattr syscall to mark such
tasks.

The scheduler may use this as hints to pack such tasks on fewer number of
cores.

Signed-off-by: Parth Shah <parth@linux.ibm.com>


---

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 84b46a992e11..0bda380d2845 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2497,3 +2497,5 @@ static inline void membarrier_switch_mm(struct rq *rq,
 {
 }
 #endif
+
+#define lenient_latency(lat)	((lat) == MAX_LATENCY_TOLERANCE)
